larpandoracontent/LArTwoDReco/HitWidthClusterMergingAlgorithm.h

```diff
@@ -0,0 +1,90 @@
+/**
+ *  @file   HitWidthClusterMergingAlgorithm.h
+ *
+ *  @brief  Header file for the hit width cluster merging algorithm class.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_HIT_WIDTH_CLUSTER_MERGING_ALGORITHM_H
+#define LAR_HIT_WIDTH_CLUSTER_MERGING_ALGORITHM_H 1
+
+#include "Pandora/Algorithm.h"
+
+#include "larpandoracontent/LArTwoDReco/LArClusterAssociation/ClusterAssociationAlgorithm.h"
+#include "larpandoracontent/LArHelpers/LArHitWidthHelper.h"
+
+namespace lar_content
+{
+
+/**
+ *  @brief HitWidthClusterMergingAlgorithm class
+ */
+class HitWidthClusterMergingAlgorithm : public ClusterAssociationAlgorithm
+{
+public:
+    /**
+     *  @brief  Default constructor
+     */
+    HitWidthClusterMergingAlgorithm();
+
+private:
+    void GetListOfCleanClusters(const pandora::ClusterList *const pClusterList, pandora::ClusterVector &clusterVector) const;
+    void PopulateClusterAssociationMap(const pandora::ClusterVector &clusterVector, ClusterAssociationMap &clusterAssociationMap) const;
+    bool IsExtremalCluster(const bool isForward, const pandora::Cluster *const pCurrentCluster,  const pandora::Cluster *const pTestCluster) const;
+
+    /**
```

Comment:
Again, no harm in it, but not necessary to doxygen comment private functions.

Reply:
I saw comments for the private functions of the CrossGaps algorithm that stems from the same base class so thought that it was Pandora style to have them in? I haven't commented the first 3 functions since they're documented in the base class

Reply:
We typically add them, although they're obviously not forming part of the publicly-accessible interface for the class. Can often help refine the code if you add/update the doxygen as you do the development.